firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/Power/AlarmsService.cpp

```diff
@@ -9,15 +9,34 @@
 
 namespace Pufferfish::Driver::Power {
 
+void AlarmsService::generate_alarm(
+    const LogEventCode &alarm_code, bool alarm_active, Application::AlarmsManager &alarms_manager) {
+  if (alarm_active) {
+    alarms_manager.activate_alarm(alarm_code, LogEventType::LogEventType_system);
+  } else {
+    alarms_manager.deactivate_alarm(alarm_code);
+  }
+}
+
 // AlarmsService
 void AlarmsService::transform(
-    const MCUPowerStatus &mcu_power_status, Application::AlarmsManager &alarms_manager) {
+    const MCUPowerStatus &mcu_power_status, Application::AlarmsManager &alarms_manager) const {
+  bool charger_disconnected = false;
   if (!mcu_power_status.charging) {
-    alarms_manager.activate_alarm(
-        LogEventCode_charger_disconnected, LogEventType::LogEventType_system);
-  } else {
-    alarms_manager.deactivate_alarm(LogEventCode_charger_disconnected);
+    charger_disconnected = true;
+  }
+  bool battery_low = false;
+  if (mcu_power_status.power_left < battery_low_power) {
+    battery_low = true;
   }
+  bool critical_battery = false;
+  if (mcu_power_status.power_left <= critical_battery_power) {
+    critical_battery = true;
+  }
```

Comment:
When we are on critical battery, does that mean that both the `battery_low` and `critical_battery` alarms are simultaneously active? The logic might be clearer if you make an enum for battery level, with `critical`, `low`, and `normal` levels, and then activate/deactivate alarms based on the enum value.

Reply:
Yeah initially i implemented that way, now I'm deactivating `battery_low` if `critical_battery` is active. and used enum class